src/App.js

```diff
@@ -1,30 +1,65 @@
-import React, { useState, useEffect } from "react";
-import fetch from "isomorphic-fetch";
-import logo from "./logo.svg";
-import "./App.css";
-
-const App = ()=>{
-	const [deals, setDeals] = useState([])
-	useEffect(()=>{
-		fetch(`http://localhost:3001/api/deals`)
-        .then(response => response.json())
-        .then(response => {
-            setDeals(response)
-        });
-	},[])
-	return (
-		<div className="App">
-        <header className="App-header">
-            <img src={logo} className="App-logo" alt="logo" />
-        </header>
-
-        <main className="BodyContent">
-            <h1>Deals <small className="text-muted">({deals.length})</small></h1>
-            <div>
-                ...
-            </div>
-        </main>
-      </div>
-	)
+import React, { useState, useEffect, useCallback } from "react"
+
+import Dropdown from "./components/Dropdown"
+import Table from "./components/Table"
+
+import { api } from "./services/api"
+import { formatDate, formatAmount } from "./utils/formaters"
+
+import logo from "./assets/logo.svg"
+import "./styles/App.css"
+
+const App = () => {
+  const [deals, setDeals] = useState([])
+  const [stats, setStats] = useState({
+    deals_count: 0,
+    total_amounts: 0,
+    avg_amount: 0,
+  })
+  const [dates, setDates] = useState([])
+  const [selected, setSelected] = useState("")
+
+  const fetchData = async () => {
+    await api.get("deals").then((res) => setDeals(res.data))
```

Comment:
If we remove `await`, will `deals` equal to `res.data` after execution?
Yes/No - Why?

Reply:
No. If we use async, it returns a promise. But I think in this case is because of how React behaves, the lifecycle methods. I'm not 100% sure on this one.

Reply:
Not entirely correct.
as it is the `await` is useless when you execute the `setDeals` into `.then()`
by writing  `await api.get("deals").then((res) => setDeals(res.data))` you are not assign any response to anything, you are just executing `setDeals` on the return of the promise
These are the two ways we can write the function fetchData to set the response data in the state deals:
```
const fetchData = () => {
   api.get("deals").then((res) => setDeals(res.data))
}
```
or
```
const fetchData = async () => {
   const res = await api.get("deals");
   setDeals(res.data);
}
```